src/ddmd/backend/elfobj.c

```diff
@@ -1691,11 +1708,73 @@ STATIC void setup_comdat(Symbol *s)
     symbol_debug(s);
     if (tyfunc(s->ty()))
     {
-        //s->Sfl = FLcode;      // was FLoncecode
-        //prefix = ".gnu.linkonce.t";   // doesn't work, despite documentation
+#if !ELF_COMDAT
         prefix = ".text.";              // undocumented, but works
         type = SHT_PROGBITS;
         flags = SHF_ALLOC|SHF_EXECINSTR;
+#else
+        reset_symbuf->write(&s, sizeof(s));
+
+        //s->Sfl = FLcode;      // was FLoncecode
+        /* ".gnu.linkonce.t" sections result in "Too many sections" errors from ld.
+         * prefix = ".gnu.linkonce.t";
+         */
+
+        /* Get section index of the comdat
+         */
+        IDXSEC comdatidx = section_cnt + 1;
```

Comment:
Bug [Issue 17338 – \[Reg 2.075\] link failure unsupported symbol section 0xff01](https://issues.dlang.org/show_bug.cgi?id=17338) happens because you're guessing the section index here, but forgot to account for extended section indices (SHT_SYMTAB_SHNDX) once section_cnt reaches SHN_LORESERVE.
I'd opt for using the newly added `addSectionToComdat` for the comdat itself, after it was actually written out to it's section (and has a correct section number).